### ISSUE TITLE
Third-party dependencies with Styled Components causes "does not have any construct or call signature" error

### ISSUE THREAD

**Issue by azer** [styled-components/styled-components#2057]
## Environment

Arch Linux, Node v10.4.1, TypeScript 3.1.1

## Reproduction

Create a TypeScript module that exports Styled Component:

```
// ./foo.ts
export default () => {
  return styled`
    background: red;
  `
}
```

Require it from another module: 

```
// ./bar.ts

import Foo from './foo'
render(<Foo />)
```

It works fine. Now make `foo` a third party dependency;

```
import Foo from 'foo'
```

You'll get `JSX element type 'Foo' does not have any construct or call signatures.`

**Comment by kachkaev** [comment]
can it be that you need to `import * as Foo from 'foo'`?

**Comment by azer** [comment]
@probablyup Right, but that wasn't the case. That's just some pseudo-code to tell the problem, as I can't share the original code. As I said, the very same code works as a local dependency does not work as a 3rd party dependency.

**Comment by probablyup** [comment]
Gotcha. I mean it has to be an issue with your setup or the code itself,
because people are using both TS and styled-components together
successfully.

**Comment by azer** [comment]
@probablyup do you know any 3rd party library based on styled-components ? I could check out and see if I'm doing something wrong.

**Comment by probablyup** [comment]
@azer maybe this would be helpful? https://github.com/patrick91/Styled-Components-Typescript-Example

**Comment by azer** [comment]
@probablyup I've seen that and I already got a working TS & SC setup. The problem is; a third party dependency on top of SC. That is the issue I mentioned above.

**Comment by probablyup** [comment]
Gotcha. Well the next thing to ask for then would be a minimal reproduction setup. It's a way to verify for both of us that it's an actual bug if you are able to cause the same error with a new project.

**Comment by azer** [comment]
@probablyup thanks for the answers. I just put up the library in a repository, this is the related line; https://github.com/azer/roka/blob/master/index.ts#L41

So, if I submodule it and import locally, like following:

```
   import roka from "./roka"

   const Container = roka()
   .stretch()
   .center()
   .sans({ size: "72px" })
   .color({ bg: "#111", fg: "#fff" })
   .div()
```

It works. But, importing it as third-party will fail with "does not have any construct or call signature" error:

```
   import roka from "roka"

   const Container = roka()
   .stretch()
   .center()
   .sans({ size: "72px" })
   .color({ bg: "#111", fg: "#fff" })
   .div()
```

**Comment by probablyup** [comment]
Hmm I have no idea. The code appears right to me? One thing I would suggest slightly offtopic with this pattern though is to make sure you're using the `css` helper as you build the style strings if you do any with interpolations.

Googling the specific error doesn't give feedback that seems actionable to what you've posted: https://stackoverflow.com/questions/31815633/what-does-the-error-jsx-element-type-does-not-have-any-construct-or-call

But maybe it'll lead you in the right direction?

**Comment by smendez-dev** [comment]
@probablyup you can reproduce the error with this lib: https://www.npmjs.com/package/react-loading-skeleton
